extensions/wikia/GlobalWatchlist/GlobalWatchlistTask.class.php

```diff
@@ -43,34 +43,50 @@ public function addWatchers( $databaseKey, $nameSpace, array $watchers ) {
 
 			// Skip revisions that doesn't exist
 			if ( !empty( $revision ) ) {
+				$watchersToAdd = [];
 				$globalWatchlistBot = new GlobalWatchlistBot();
-
-				$db = wfGetDB( DB_MASTER, [ ], \F::app()->wg->ExternalDatawareDB );
 				foreach ( $watchers as $watcherID ) {
 					if ( $globalWatchlistBot->shouldNotSendDigest( $watcherID ) ) {
-						$this->clearGlobalWatchlistAll( $watcherID );
 						continue;
 					}
-
-					( new WikiaSQL() )
-						->INSERT()->INTO( GlobalWatchlistTable::TABLE_NAME )
-						->SET( GlobalWatchlistTable::COLUMN_USER_ID, $watcherID )
-						->SET( GlobalWatchlistTable::COLUMN_CITY_ID, \F::app()->wg->CityId )
-						->SET( GlobalWatchlistTable::COLUMN_TITLE, $databaseKey )
-						->SET( GlobalWatchlistTable::COLUMN_NAMESPACE, $nameSpace )
-						->SET( GlobalWatchlistTable::COLUMN_REVISION_ID, $revision->getId() )
-						->SET( GlobalWatchlistTable::COLUMN_REVISION_TIMESTAMP, $revision->getTimestamp() )
-						->SET( GlobalWatchlistTable::COLUMN_TIMESTAMP, $revision->getTimestamp() )
-						// Do nothing on duplicate key - we already have that record in place
-						->ON_DUPLICATE_KEY_UPDATE(
-							[ GlobalWatchlistTable::COLUMN_USER_ID => $watcherID ]
-						)
-						->run( $db );
+					$watchersToAdd[] = [
+						$watcherID,
+						\F::app()->wg->CityId,
+						$databaseKey,
+						$nameSpace,
+						$revision->getId(),
+						$revision->getTimestamp(),
+						$revision->getTimestamp()
+					];
 				}
+
+				$this->addWatchersToDb( $watchersToAdd );
 			}
 		}
 	}
 
+	private function addWatchersToDb( array $watchers ) {
+		$columns = [
+			GlobalWatchlistTable::COLUMN_USER_ID,
+			GlobalWatchlistTable::COLUMN_CITY_ID,
+			GlobalWatchlistTable::COLUMN_TITLE,
+			GlobalWatchlistTable::COLUMN_NAMESPACE,
+			GlobalWatchlistTable::COLUMN_REVISION_ID,
+			GlobalWatchlistTable::COLUMN_REVISION_TIMESTAMP,
+			GlobalWatchlistTable::COLUMN_TIMESTAMP
+		];
+
+		$db = wfGetDB( DB_MASTER, [ ], \F::app()->wg->ExternalDatawareDB );
+		( new WikiaSQL() )
+			->INSERT()->INTO( GlobalWatchlistTable::TABLE_NAME, $columns )
+			->VALUES( $watchers )
+			// Do nothing on duplicate key - we already have that record in place
+			->ON_DUPLICATE_KEY_UPDATE(
+				[ GlobalWatchlistTable::COLUMN_TIMESTAMP => GlobalWatchlistTable::COLUMN_TIMESTAMP ]
```

Comment:
I changed this from `GlobalWatchlistTable::COLUMN_USER_ID` to `GlobalWatchlistTable::COLUMN_TIMESTAMP`. Since we're doing a bulk write, we don't have access to the $watcherId value like we did before. I tried updating to 
```PHP
[ GlobalWatchlistTable::COLUMN_USER_ID => GlobalWatchlistTable::COLUMN_USER_ID ]
```
and it was throwing the following error `ERROR 1062 (23000): Duplicate entry '869155-0-0-FBPlugins' for key 'wikia_user'`. Updating this to use a non-indexed key fixes it and since this is essentially a no-op, I think it should be fine for our needs.